vortex-array/src/ptype.rs

```diff
@@ -139,6 +139,22 @@ macro_rules! match_each_integer_ptype {
 }
 pub use match_each_integer_ptype;
 
+#[macro_export]
+macro_rules! match_each_float_ptype {
+    ($self:expr, | $_:tt $enc:ident | $($body:tt)*) => ({
+        macro_rules! __with__ {( $_ $enc:ident ) => ( $($body)* )}
+        use $crate::ptype::PType;
+        use half::f16;
+        match $self {
+            PType::F16 => __with__! { f16 },
+            PType::F32 => __with__! { f32 },
+            PType::F64 => __with__! { f64 },
+            _ => panic!("Unsupported ptype {}", $self),
+        }
+    })
+}
+pub use match_each_float_ptype;
```

Comment:
I've realised these `pub use` statements aren't actually necessary. Could you get rid of them from this file?